// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package eratos.blocks;

// The greeting service definition.
service BlockEventV1 {
  // Sends a block event to register
  rpc RegisterBlockEvent (RegisterBlockRequest) returns (RegisterBlockReply) {}
  // Sends a block event to log
  rpc LogBlockEvent (RegisterBlockRequest) returns (RegisterBlockReply) {}
  // Report a block event to server
  rpc ReportBlockEvent (RegisterBlockRequest) returns (RegisterBlockReply) {}  
}

// The request message containing the user's name.
message RegisterBlockRequest {
  string name = 1;
}

// The response message containing the greetings
message RegisterBlockReply {
  string message = 1;
}
